File: src/main/java/ru/citeck/ecos/settings/EcosServerCollectionItemEditor.java
package ru.citeck.ecos.settings;

import com.intellij.util.ui.CollectionItemEditor;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

class EcosServerCollectionItemEditor implements CollectionItemEditor<EcosServer> {

    @Override
    public @NotNull Class<? extends EcosServer> getItemClass() {
        return EcosServer.class;
    }

    @Override
    public EcosServer clone(@NotNull EcosServer item, boolean forInPlaceEditing) {
        EcosServer clone = item.clone();
        clone.setId(UUID.randomUUID().toString());
        return clone;
    }

}

File: src/main/kotlin/ru/citeck/idea/settings/servers/EcosServerCollectionItemEditor.kt
package ru.citeck.idea.settings.servers

import com.intellij.util.ui.CollectionItemEditor
import java.util.*

internal class EcosServerCollectionItemEditor : CollectionItemEditor<CiteckServer.Builder> {

    override fun getItemClass(): Class<out CiteckServer.Builder> {
        return CiteckServer.Builder::class.java
    }

    override fun clone(item: CiteckServer.Builder, forInPlaceEditing: Boolean): CiteckServer.Builder {
        return item.build().copy().withId(UUID.randomUUID().toString())
    }
}
